### PR TITLE
printf/scanf format checks

### PR DESCRIPTION
**Pull request by yamt**
## Summary
printflike stuff separated from https://github.com/apache/incubator-nuttx/pull/2222

### CODE REVIEW

File: include/stdio.h [hunk 1]
```diff
@@ -174,28 +176,36 @@ int    ungetc(int c, FAR FILE *stream);
...
+int    scanf(FAR const IPTR char *fmt, ...) scanflike(1, 2);
+int    vasprintf(FAR char **ptr, FAR const IPTR char *fmt, va_list ap)
+       printflike(2, 0);
 int    vfprintf(FAR FILE *stream, FAR const IPTR char *fmt,
-         va_list ap);
-int    vfscanf(FAR FILE *stream, FAR const IPTR char *fmt, va_list ap);
-int    vprintf(FAR const IPTR char *fmt, va_list ap);
-int    vscanf(FAR const IPTR char *fmt, va_list ap);
+         va_list ap)
+       printflike(2, 0);
```
Comment (xiaoxiang781216) [inline include/stdio.h:198]: merge to line 197?

**Comment by xiaoxiang781216** [comment]
Does it make more sense to merge into one patch or two(one for printflike, another scanflike)?

File: include/stdio.h (hunk 1 above)
Comment (yamt) [reply include/stdio.h]: done

**Comment by yamt** [comment]
> Does it make more sense to merge into one patch or two(one for printflike, another scanflike)?

for me, smaller patches are easier to manage. (revert, cherry-pick, etc)

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**